crates/aquascope/src/interpreter/mvalue.rs

```diff
@@ -515,3 +517,27 @@ impl<'tcx> VisEvaluator<'_, 'tcx> {
     .read(op)
   }
 }
+
+// trait InterpCxExt<'tcx> {
+//   fn check_mplace(
+//     &self,
+//     mplace: &MPlaceTy<'tcx, miri::Provenance>,
+//   ) -> InterpResult<'tcx>;
+// }
+
+// impl<'mir, 'tcx> InterpCxExt<'tcx>
+//   for miri::InterpCx<'mir, 'tcx, miri::MiriMachine<'mir, 'tcx>>
+// {
+//   fn check_mplace(
+//     &self,
+//     mplace: &MPlaceTy<'tcx, miri::Provenance>,
+//   ) -> InterpResult<'tcx> {
+//     let (size, _align) = self
+//       .size_and_align_of_mplace(&mplace)?
+//       .unwrap_or((mplace.layout.size, mplace.layout.align.abi));
+//     // Due to packed places, only `mplace.align` matters.
+//     let align = if M::enforce_alignment(self) { mplace.align } else { Align::ONE };
+//     self.check_ptr_access_align(mplace.ptr(), size, align, miri::CheckInAllocMsg::MemoryAccessTest)?;
```

Comment:
This was the old code for `check_mplace` in rustc_const_eval. (except the `CheckInAllocMsg` type was `PtrDeref` (or something of that flavor, the invariant was removed)